src/Sentry/Scope.cs

```diff
@@ -99,13 +99,31 @@ public Contexts Contexts
             set => _contexts = value;
         }
 
+        // Internal for testing.
+        internal Action UserChanged => () =>
+        {
+            if (Options.EnableScopeSync &&
+                Options.ScopeObserver is { } observer)
+            {
+                observer.User = User;
```

Comment:
Can we change this action to take the user as a reference? Technically the User being read here could be different than the one set if they raced (global hub mode)